src/ships.js

```diff
@@ -0,0 +1,33 @@
+function Ship(itinerary) {
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.setup()
+};
+
+Ship.prototype = {
+setSail() {
+    const itinerary = this.itinerary;
+    const currentPortIndex = itinerary.ports.indexOf(this.currentPort)
+
+    if(currentPortIndex === (itinerary.ports.length -1)) {
+        throw new Error('End of itinerary reached');
+    };
+    this.previousPort = this.currentPort;
+    this.currentPort.removeShip(this)
+    this.currentPort = null; 
+},
+dock() {
+    const itinerary = this.itinerary;
+    const previousPortIndex = itinerary.ports.indexOf(this.previousPort);
+
+    this.currentPort = itinerary.ports[previousPortIndex, +1];
+    this.currentPort.addShip(this);
+},
+setup() {
+    
+    this.currentPort.addShip(this)
+}
```

Comment:
This should all be indented 4 spaces, plus the blank line inside the `setup` function should be removed